api/v1beta2/configuration_types.go

```diff
@@ -104,12 +106,393 @@ type Property struct {
 	Value string `json:"value,omitempty"`
 }
 
-// Backend stores the state in a Kubernetes secret with locking done using a Lease resource.
+// BackendType is an enum type for the Terraform backend types
+// +kubebuilder:validation:Enum=remote;artifactory;azurerm;consul;cos;etcd;etcdv3;gcs;http;kubernetes;manta;oso;pg;s3;swift
+type BackendType string
+
+// Backend describes the Terraform backend configuration
 type Backend struct {
 	// SecretSuffix used when creating secrets. Secrets will be named in the format: tfstate-{workspace}-{secretSuffix}
 	SecretSuffix string `json:"secretSuffix,omitempty"`
 	// InClusterConfig Used to authenticate to the cluster from inside a pod. Only `true` is allowed
 	InClusterConfig bool `json:"inClusterConfig,omitempty"`
+
+	// Inline allows users to use raw hcl code to specify their Terraform backend
+	Inline string `json:"inline,omitempty"`
+
+	// BackendType indicates which backend type to use. This field is needed for custom backend configuration.
+	BackendType BackendType `json:"backendType,omitempty"`
+
+	// Remote is needed for the Terraform `remote` backend type.
+	Remote *RemoteBackendConf `json:"remote,omitempty"`
+
+	// Artifactory is needed for the Terraform `artifactory` backend type.
+	Artifactory *ArtifactoryBackendConf `json:"artifactory,omitempty"`
+
+	// Azurerm is needed for the Terraform `azurerm` backend type.
+	Azurerm *AzurermBackendConf `json:"azurerm,omitempty"`
+
+	// Consul is needed for the Terraform `consul` backend type.
+	Consul *ConsulBackendConf `json:"consul,omitempty"`
+
+	// Consul is needed for the Terraform `consul` backend type.
+	COS *COSBackendConf `json:"cos,omitempty"`
+
+	// ETCD is needed for the Terraform `etcd` backend type.
+	ETCD *ETCDBackendConf `json:"etcd,omitempty"`
+
+	// ETCDV3 is needed for the Terraform `etcdv3` backend type.
+	ETCDV3 *ETCDV3BackendConf `json:"etcdv3,omitempty"`
+
+	// GCS is needed for the Terraform `gcs` backend type.
+	GCS *GCSBackendConf `json:"gcs,omitempty"`
+
+	// HTTP is needed for the Terraform `http` backend type.
+	HTTP *HTTPBackendConf `json:"http,omitempty"`
+
+	// Kubernetes is needed for the Terraform `kubernetes` backend type.
+	Kubernetes *KubernetesBackendConf `json:"kubernetes,omitempty"`
+
+	// Manta is needed for the Terraform `manta` backend type.
+	Manta *MantaBackendConf `json:"manta,omitempty"`
+
+	// OSS is needed for the Terraform `oss` backend type.
+	OSS *OSSBackendConf `json:"oss,omitempty"`
+
+	// PG is needed for the Terraform `pg` backend type.
+	PG *PGBackendConf `json:"pg,omitempty"`
+
+	// s3 is needed for the Terraform `s3` backend type.
+	S3 *S3BackendConf `json:"s3,omitempty"`
+
+	// Swift is needed for the Terraform `swift` backend type.
+	Swift *SwiftBackendConf `json:"swift,omitempty"`
+}
+
+// CurrentNSSecretSelector is used to specify the key in a secret in the current namespace.
+type CurrentNSSecretSelector struct {
```

Comment:
What's the current namespace?

Reply:
The same namespace as the configuration.